File: Archiver/CS/clsRECIPIENTS.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Linq;
using System.Diagnostics;
using System.Data;
using Microsoft.VisualBasic;
using MODI;
using System.Xml.Linq;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.Configuration;
using System.Data.SqlClient;
using System.IO;
using System.Data.Sql;



namespace EcmArchiveClcSetup
{
	public class clsRECIPIENTS
	{
		
		
		//** DIM the selected table columns
		clsDatabase DB = new clsDatabase();
		clsDma DMA = new clsDma();
		clsUtility UTIL = new clsUtility();
		clsLogging LOG = new clsLogging();
		
		
		
		string Recipient = "";
		string EmailGuid = "";
		string TypeRecp = "";
		
		
		
		
		//** Generate the SET methods
		public void setRecipient(ref string val)
		{
			if (val.Length == 0)
			{
				MessageBox.Show("SET: Field \'Recipient\' cannot be NULL.");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			Recipient = val;
		}
		
		
		public void setEmailguid(ref string val)
		{
			if (val.Length == 0)
			{
				MessageBox.Show("SET: Field \'Emailguid\' cannot be NULL.");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			EmailGuid = val;
		}
		
		
		public void setTyperecp(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			TypeRecp = val;
		}
		
		
		
		
		
		
		//** Generate the GET methods
		public string getRecipient()
		{
			if (Recipient.Length == 0)
			{
				MessageBox.Show("GET: Field \'Recipient\' cannot be NULL.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(Recipient);
		}
		
		
		public string getEmailguid()
		{
			if (EmailGuid.Length == 0)
			{
				MessageBox.Show("GET: Field \'Emailguid\' cannot be NULL.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(EmailGuid);
		}
		
		
		public string getTyperecp()
		{
			return UTIL.RemoveSingleQuotes(TypeRecp);
		}
		
		
		
		
		
		
		//** Generate the Required Fields Validation method
		public bool ValidateReqData()
		{
			if (Recipient.Length == 0)
			{
				return false;
			}
			if (EmailGuid.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		
		
		//** Generate the Validation method
		public bool ValidateData()
		{
			if (Recipient.Length == 0)
			{
				return false;
			}
			if (EmailGuid.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		
		
		//** Generate the INSERT method
		public bool Insert()
		{
			bool b = false;
			string s = "";
			s = s + " INSERT INTO Recipients(";
			s = s + "Recipient,";
			s = s + "EmailGuid,";
			s = s + "TypeRecp) values (";
			s = s + "\'" + Recipient + "\'" + ",";
			s = s + "\'" + EmailGuid + "\'" + ",";
			s = s + "\'" + TypeRecp + "\'" + ")";
			return DB.ExecuteSqlNewConn(s, false);
			
			
		}
		
		
		
		
		//** Generate the UPDATE method
		public bool Update(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			
			s = s + " update Recipients set ";
			s = s + "Recipient = \'" + getRecipient() + "\'" + ", ";
			s = s + "EmailGuid = \'" + getEmailguid() + "\'" + ", ";
			s = s + "TypeRecp = \'" + getTyperecp() + "\'";
			WhereClause = (string) (" " + WhereClause);
			s = s + WhereClause;
			return DB.ExecuteSqlNewConn(s, false);
		}
		
		
		
		
		//** Generate the SELECT method
		public SqlDataReader SelectRecs()
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "Recipient,";
			s = s + "EmailGuid,";
			s = s + "TypeRecp ";
			s = s + " FROM Recipients";
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		
		
		//** Generate the Select One Row method
		public SqlDataReader SelectOne(string WhereClause)
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "Recipient,";
			s = s + "EmailGuid,";
			s = s + "TypeRecp ";
			s = s + " FROM Recipients";
			s = s + WhereClause;
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		
		
		//** Generate the DELETE method
		public bool Delete(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			
			WhereClause = (string) (" " + WhereClause);
			
			
			s = " Delete from Recipients";
			s = s + WhereClause;
			
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
			
		}
		
		
		
		
		//** Generate the Zeroize Table method
		public bool Zeroize()
		{
			bool b = false;
			string s = "";
			
			
			s = s + " Delete from Recipients";
			
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
			
		}
		
		
		
		
		//** Generate Index Queries
		public int cnt_PK32A(string EmailGuid, string Recipient)
		{
			
			
			Recipient = UTIL.RemoveSingleQuotes(Recipient);
			
			
			int B = 0;
			string TBL = "Recipients";
			string WC = "Where EmailGuid = \'" + EmailGuid + "\' and   Recipient = \'" + Recipient + "\'";
			
			
			B = DB.iGetRowCount(TBL, WC);
			
			
			return B;
		} //** cnt_PK32A
		
		
		//** Generate Index ROW Queries
		public SqlDataReader getRow_PK32A(string EmailGuid, string Recipient)
		{
			
			
			SqlDataReader rsData = null;
			string TBL = "Recipients";
			string WC = "Where EmailGuid = \'" + EmailGuid + "\' and   Recipient = \'" + Recipient + "\'";
			
			
			rsData = DB.GetRowByKey(TBL, WC);
			
			
			if (rsData.HasRows)
			{
				return rsData;
			}
			else
			{
				return null;
			}
		} //** getRow_PK32A
		
		
		/// Build Index Where Caluses
		///
		public string wc_PK32A(string EmailGuid, string Recipient)
		{
			
			
			string WC = "Where EmailGuid = \'" + EmailGuid + "\' and   Recipient = \'" + Recipient + "\'";
			
			
			return WC;
		} //** wc_PK32A
	}
	
}

File: SearchSVC/clsRECIPIENTS.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql


Public Class clsRECIPIENTS


    '** DIM the selected table columns 
    Dim DB As New clsDatabaseSVR
    Dim DMA As New clsDmaSVR
    Dim UTIL As New clsUtilitySVR
    Dim LOG As New clsLogging



    Dim Recipient As String = ""
    Dim EmailGuid As String = ""
    Dim TypeRecp As String = ""


    Dim SecureID As Integer = -1

    Sub New(ByVal iSecureID As Integer)
        SecureID = iSecureID
    End Sub


    '** Generate the SET methods 
    Public Sub setRecipient(ByRef val as string)
        If Len(val) = 0 Then
            MsgBox("SET: Field 'Recipient' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        Recipient = val
    End Sub


    Public Sub setEmailguid(ByRef val as string)
        If Len(val) = 0 Then
            MsgBox("SET: Field 'Emailguid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        EmailGuid = val
    End Sub


    Public Sub setTyperecp(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        TypeRecp = val
    End Sub






    '** Generate the GET methods 
    Public Function getRecipient() As String
        If Len(Recipient) = 0 Then
            MsgBox("GET: Field 'Recipient' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(Recipient)
    End Function


    Public Function getEmailguid() As String
        If Len(EmailGuid) = 0 Then
            MsgBox("GET: Field 'Emailguid' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(EmailGuid)
    End Function


    Public Function getTyperecp() As String
        Return UTIL.RemoveSingleQuotes(TypeRecp)
    End Function






    '** Generate the Required Fields Validation AS string 
    Public Function ValidateReqData() As Boolean
        If Recipient.Length = 0 Then Return false
        If EmailGuid.Length = 0 Then Return false
        Return True
    End Function




    '** Generate the Validation AS string 
    Public Function ValidateData() As Boolean
        If Recipient.Length = 0 Then Return false
        If EmailGuid.Length = 0 Then Return false
        Return True
    End Function




    '** Generate the INSERT AS string 
    Public Function Insert() As Boolean
        Dim b As Boolean = false
        Dim s As String = ""
        s = s + " INSERT INTO Recipients("
        s = s + "Recipient,"
        s = s + "EmailGuid,"
        s = s + "TypeRecp) values ("
        s = s + "'" + Recipient + "'" + ","
        s = s + "'" + EmailGuid + "'" + ","
        s = s + "'" + TypeRecp + "'" + ")"
        Return DB.DBExecuteSql(SecureID, S, False)


    End Function




    '** Generate the UPDATE AS string 
    Public Function Update(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        s = s + " update Recipients set "
        s = s + "Recipient = '" + getRecipient() + "'" + ", "
        s = s + "EmailGuid = '" + getEmailguid() + "'" + ", "
        s = s + "TypeRecp = '" + getTyperecp() + "'"
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DB.DBExecuteSql(SecureID, S, False)
    End Function




    '** Generate the SELECT AS string 
    Public Function SelectRecs(ByVal SecureID As Integer) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "Recipient,"
        s = s + "EmailGuid,"
        s = s + "TypeRecp "
        s = s + " FROM Recipients"

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the Select One Row AS string 
    Public Function SelectOne(ByVal SecureID As Integer, ByVal WhereClause as string) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "Recipient,"
        s = s + "EmailGuid,"
        s = s + "TypeRecp "
        s = s + " FROM Recipients"
        s = s + WhereClause

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the DELETE AS string 
    Public Function Delete(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        WhereClause = " " + WhereClause


        s = " Delete from Recipients"
        s = s + WhereClause


        b = DB.DBExecuteSql(SecureID, S, False)
        Return b


    End Function




    '** Generate the Zeroize Table AS string 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        s = s + " Delete from Recipients"


        b = DB.DBExecuteSql(SecureID, S, False)
        Return b


    End Function




    '** Generate Index Queries 
    Public Function cnt_PK32A(ByVal EmailGuid As String, ByVal Recipient As String) As Integer


        Recipient = UTIL.RemoveSingleQuotes(Recipient)


        Dim B As Integer = 0
        Dim TBL$ = "Recipients"
        Dim WC$ = "Where EmailGuid = '" + EmailGuid + "' and   Recipient = '" + Recipient + "'"


        B = DB.iGetRowCount(TBL$, WC)


        Return B
    End Function     '** cnt_PK32A


    '** Generate Index ROW Queries 
    Public Function getRow_PK32A(ByVal SecureID As Integer, ByVal EmailGuid As String, ByVal Recipient As String) As SqlDataReader


        Dim rsData As SqlDataReader = Nothing
        Dim TBL$ = "Recipients"
        Dim WC$ = "Where EmailGuid = '" + EmailGuid + "' and   Recipient = '" + Recipient + "'"


        rsData = DB.GetRowByKey(SecureID, TBL$, WC)


        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK32A


    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK32A(ByVal EmailGuid As String, ByVal Recipient As String) As String


        Dim WC$ = "Where EmailGuid = '" + EmailGuid + "' and   Recipient = '" + Recipient + "'"


        Return WC
    End Function     '** wc_PK32A
End Class
